### PR TITLE
 Integrate VAM TS 

### PR DESCRIPTION
**Pull request by gkueppers**
**Adding VAM TS into etsi_its_messages**

- adapt launch.json
- add message-package
- add coding-package
- add conversion package
- run all conversions
- integrate VAM into conversion node
- adapt README
- integrate VAM into gitlab- and github CI-Pipeline
- add publish script for VAM
- add parameter to disable asn-constraint-check before encoding of asn1 struct (check could lead to segmentation-fault because of [infinte recursion](https://github.com/vlm/asn1c/issues/410))

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,6 +165,7 @@ rosrun nodelet nodelet standalone etsi_its_conversion/Converter _etsi_types:=[ca
 | `~/cam_ts/in` | `etsi_its_cam_ts_msgs/msg/CAM` | CAM (TS) for conversion to UDP |
 | `~/cpm_ts/in` | `etsi_its_cpm_ts_msgs/msg/CollectivePerceptionMessage` | CPM for conversion to UDP |
 | `~/denm/in` | `etsi_its_denm_msgs/msg/DENM` | DENM for conversion to UDP |
+| `~/vam_ts/in` | `etsi_its_vam_ts_msgs/msg/CAM` | VAM (TS) for conversion to UDP |
```
Comment (lreiher) [inline README.md:168]: Wrong type

File: README.md [hunk 2]
```diff
@@ -171,6 +176,7 @@ rosrun nodelet nodelet standalone etsi_its_conversion/Converter _etsi_types:=[ca
 | `~/cam_ts/out` | `etsi_its_cam_ts_msgs/msg/CAM` | CAM (TS) converted from UDP payload |
 | `~/cpm_ts/out` | `etsi_its_cpm_ts_msgs/msg/CollectivePerceptionMessage` | CPM converted from UDP payload |
 | `~/denm/out` | `etsi_its_denm_msgs/msg/DENM` | DENM converted from UDP payload |
+| `~/vam_ts/out` | `etsi_its_vam_ts_msgs/msg/CAM` | VAM (TS) converted from UDP payload |
```
Comment (lreiher) [inline README.md:179]: Wrong type

**Comment by jpbusch** [comment]
Please add a minimal example for publishing a VAM (like [here](https://github.com/ika-rwth-aachen/etsi_its_messages/blob/main/utils/scripts/publish_cam.py)). This allows us to test the encoding and decoding before merging.

**Comment by gkueppers** [comment]
> Please add a minimal example for publishing a VAM (like [here](https://github.com/ika-rwth-aachen/etsi_its_messages/blob/main/utils/scripts/publish_cam.py)). This allows us to test the encoding and decoding before merging.

Currently the conversion node is crashing with a segmentation fault.
Reason for this is an infinite recursion of `vam_ts_ItsPduHeaderVam_constraint` in [Line 26 of vam_ts_ItsPduHeaderVam.c](https://github.com/ika-rwth-aachen/etsi_its_messages/blob/fdd54440f51e8d3bf72cb5b3576af57c25d07849/etsi_its_coding/etsi_its_vam_ts_coding/src/vam_ts_ItsPduHeaderVam.c#L26).

It seems like there is a bug in asn1c which is may related to [this issue](https://github.com/vlm/asn1c/issues/410).

**Comment by gkueppers** [comment]
> > Please add a minimal example for publishing a VAM (like [here](https://github.com/ika-rwth-aachen/etsi_its_messages/blob/main/utils/scripts/publish_cam.py)). This allows us to test the encoding and decoding before merging.
> 
> Currently the conversion node is crashing with a segmentation fault. Reason for this is an infinite recursion of `vam_ts_ItsPduHeaderVam_constraint` in [Line 26 of vam_ts_ItsPduHeaderVam.c](https://github.com/ika-rwth-aachen/etsi_its_messages/blob/fdd54440f51e8d3bf72cb5b3576af57c25d07849/etsi_its_coding/etsi_its_vam_ts_coding/src/vam_ts_ItsPduHeaderVam.c#L26).
> 
> It seems like there is a bug in asn1c which is may related to [this issue](https://github.com/vlm/asn1c/issues/410).

"Fixed" by disabling the asn-constraint check. Encoding still fails with an meaningful error-message if you try to encode an asn1-struct that does not conform to the specified asn-constraints.

File: etsi_its_conversion/etsi_its_conversion/src/Converter.cpp [hunk 3]
```diff
@@ -96,6 +96,8 @@ const std::string Converter::kSubscriberQueueSizeParam{"subscriber_queue_size"};
 const int Converter::kSubscriberQueueSizeParamDefault{10};
 const std::string Converter::kPublisherQueueSizeParam{"publisher_queue_size"};
 const int Converter::kPublisherQueueSizeParamDefault{10};
+const std::string Converter::kCheckConstraintsBeforeEncodingParam{"check_constraints_before_encoding"};
+const bool Converter::kCheckConstraintsBeforeEncodingParamDefault{true};
```
Comment (jpbusch) [inline etsi_its_conversion/etsi_its_conversion/src/Converter.cpp:100]: also default to `false` here

File: utils/scripts/README.md [hunk 4]
```diff
@@ -11,7 +11,8 @@ ros2 run etsi_its_conversion etsi_its_conversion_node \
         -r /etsi_its_conversion/udp/out:=/etsi_its_conversion/udp/in \
         -p has_btp_destination_port:=true \
         -p btp_destination_port_offset:=0 \
-        -p etsi_message_payload_offset:=4
+        -p etsi_message_payload_offset:=4 \
+        -p check_constraints_before_encoding:=false
```
Comment (jpbusch) [inline utils/scripts/README.md:15]: should not be necessary if `check_constraints_before_encoding` defaults to `false`

**Review by jpbusch — APPROVED** [review]

**Pull request merged**